File: Sources/SnipKeyApp/ClipboardSnippetFactory.swift
import Foundation
import SnipKeyCore

enum ClipboardSnippetFactory {
    static func makeSnippet(from content: String, existingSnippets: [Snippet]) -> Snippet {
        Snippet(
            trigger: SnippetTriggerSuggester.suggestTrigger(
                for: content,
                existingTriggers: existingSnippets.map(\ .trigger)
            ),
            replacement: content
        )
    }
}

File: Windows/SnipKey.Windows/UI/ClipboardSnippetFactory.cs
using SnipKey.WinApp.Core;

namespace SnipKey.WinApp.UI;

internal static class ClipboardSnippetFactory
{
    public static Snippet MakeSnippet(string content, IEnumerable<Snippet> existingSnippets)
    {
        return new Snippet
        {
            Trigger = SnippetTriggerSuggester.SuggestTrigger(content, existingSnippets.Select(snippet => snippet.Trigger)),
            Replacement = content
        };
    }
}